### DISCUSSION TITLE
Are the Accelerometer Biases  automatically calibrated?

### DISCUSSION THREAD

**Discussion by yonimandel-spearuav** [bitcraze/discussions#1279] [Q&A]
Are the Accelerometer Biases  automatically calibrated?
If so, where is it in the code?
Thanks

**Comment by ToveRumar** [comment]
Hi!
Thank you for your question!

We do not do/support doing a full 3-axis calibration of the accelerometer vectors, where the Crazyflie is rotated in 3D space. 

However, at startup the accelerometer z-bias is calculated and that is used to also calibrate all three axes.

The IMU is calibrated in `src/hal/src/sensors_bmi088_bmp3xx.c`.
Hope this answered your question.

**Reply by yonimandel-spearuav** [reply]
Hi Tove,
Where in the code is the accelerometer z-bias calculated? Couldn't find it.
Thanks, Yoni

**Reply by ToveRumar** [reply]
Hi!

In this file it is called "scale" and not "remove bias". It can be found in this function;

`processAccScale`

This is where we find the scale to use, which also can be understood as bias calibration.
Is this what you were looking for?
